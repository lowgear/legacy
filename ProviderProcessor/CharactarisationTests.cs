using System;
using NUnit.Framework;
using ApprovalTests;
using ApprovalTests.Reporters;
using ProviderProcessing.ProviderDatas;
using FakeItEasy;
using Newtonsoft.Json;
using ProviderProcessing.References;

namespace ProviderProcessing
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class CharactarisationTests
    {
        private ProviderProcessor pp;
        private ProviderRepository fakeRepo;
        private ProviderData data = new ProviderData();
        private ProductsReference fakeReference = A.Fake<ProductsReference>();

        [SetUp]
        public void SetUp()
        {
            fakeRepo = A.Fake<ProviderRepository>();

//            pp = new ProviderProcessor(fakeRepo);
            pp = A.Fake<ProviderProcessor>(op => op.WithArgumentsForConstructor(new [] {fakeRepo}));
        }

        [Test]
        public void DoSomething_WhenSomething()
        {
            A.CallTo(() => fakeRepo.FindByProviderId(A<Guid>._)).Returns(data);
            A.CallTo(() => pp.GetInstance()).Returns(fakeReference);
            A.CallTo(() => fakeReference.FindCodeByName(A<string>._)).Returns(0);
            data.Products = new ProductData[0];

            var message = JsonConvert.SerializeObject(data);

            Approvals.Verify(pp.ProcessProviderData(message));
        }
    }
}
