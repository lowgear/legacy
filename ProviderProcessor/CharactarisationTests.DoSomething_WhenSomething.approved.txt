Success: True, Error: OK, Most Critical Message:
